iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java

```diff
@@ -179,9 +179,17 @@ public ParameterDeclaration parameterDeclaration(
     SyntaxToken keyword,
     Identifier name,
     TypeExpressionAble typeExpression,
-    Optional<SyntaxToken> equ,
-    Optional<Expression> defaultValue) {
-    return new ParameterDeclarationImpl(decorators.or(emptyList()), keyword, name, typeExpression, equ.orNull(), defaultValue.orNull());
+    Optional<Tuple<SyntaxToken, Expression>> equDefaultValue) {
+    if (equDefaultValue.isPresent()) {
+      return new ParameterDeclarationImpl(decorators.or(emptyList()),
+        keyword,
+        name,
+        typeExpression,
+        equDefaultValue.get().first(),
+        equDefaultValue.get().second());
+    } else {
+      return new ParameterDeclarationImpl(decorators.or(emptyList()), keyword, name, typeExpression, null, null);
+    }
```

Comment:
This is moving a bit of the grammar logic into the `TreeFactory`, which could be handled on the grammar side with a `b.firstOf` and the two different forms.
Again not sure what's the involved complexity/performance behind, probably something we could discuss in the incoming presentation of Nils.
I'm ok to merge this as it is.